sway-ir/src/parser.rs

```diff
@@ -52,16 +52,34 @@ mod ir_builder {
                 }
 
             rule fn_decl() -> IrAstFnDecl
-                = "fn" _ name:id() _ selector:selector_id()? _ "(" _
-                      args:(block_arg() ** comma()) ")" _ "->" _ ret_type:ast_ty()
-                          metadata:comma_metadata_idx()? "{" _
-                      locals:fn_local()*
-                      blocks:block_decl()*
-                  "}" _ {
+                = "pub fn" _ name:id() _ selector:selector_id()? _ "(" _
```

Comment:
I don't feel that the IR should care about things like `pub`. In my view, `pub` is only relevant for the frontend, similarly to `mut`. After they're consumed in the frontend, `pub` and `mut` become obsolete and they should not really carry any weight in IR. I'll let @otrho make the final call on this though.